src/checkovInstaller.ts

```diff
@@ -1,25 +1,48 @@
+import * as path from 'path';
 import { Logger } from 'winston';
 import { asyncExec } from './utils';
 
-const updateCheckovWithPip3 = async (logger: Logger): Promise<void> => {
+const installOrUpdateCheckovWithPip3 = async (logger: Logger): Promise<string> => {
     try {
         logger.info('Trying to install Checkov using pip3.');
-        await asyncExec('pip3 install -U checkov -i https://pypi.org/simple/');
-        logger.info('Checkov installed successfully using pip3.');
+        await asyncExec('pip3 install -U --user checkov -i https://pypi.org/simple/');
+        const [pythonUserBaseOutput] = await asyncExec('python3 -c \'import site; print(site.USER_BASE)\'');
+        const checkovPath = path.join(pythonUserBaseOutput.trim(), 'bin', 'checkov');
+        logger.info('Checkov installed successfully using pip3.', { checkovPath });
+        return checkovPath;
     } catch (error) {
         logger.error('Failed to install or update Checkov using pip3. Error:', { error });
         throw new Error('Failed to install or update Checkov using pip3');
     }
 };
 
-type CheckovPython = 'pip3';
+const isCheckovInstalledBeforePlugin = async () => {
+    try {
+        await asyncExec('checkov --version');
+        return true;
+    } catch (err) {
+        return false;
+    }
+};
+
+type CheckovInstallationMethod = 'pip3' | 'unknown';
 export interface CheckovInstallation {
-    checkovPython: CheckovPython;
+    checkovInstallationMethod: CheckovInstallationMethod;
+    checkovPath: string;
 }
 
 export const installOrUpdateCheckov = async (logger: Logger): Promise<CheckovInstallation> => {
+    const checkovInstalledBeforePlugin = await isCheckovInstalledBeforePlugin();
+    if (checkovInstalledBeforePlugin) {
+        logger.info('Checkov is installed on machine before plugin.');
```

Comment:
but what if it's not from the right version? users would not receive the value? how do you know it's not working for existing checkov users?